compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -2176,4 +2176,8 @@ trait Applications extends Compatibility {
       report.error(em"not an extension method: $methodRef", receiver.srcPos)
     app
   }
+
+  def isApplicableExtensionMethod(ref: TermRef, receiver: Type)(using Context) =
```

Comment:
You could check `!receiver.isBottomType` here to exclude Nothing and Null so you don't have to exclude them in the caller.

Reply:
Right, I can move the check here but not sure we should exclude Null. The compiler still lets you define an extension on Null so I think for consistency this should be displayed in a completion

Reply:
> The compiler still lets you define an extension on Null so I think for consistency this should be displayed in a completion

I don't think it's a big deal either way, but because by default `Null` is a subtype of every reference type, one should be careful to not include all the extension methods that apply to reference types in the list

Reply:
Checking `isBottomType` then